src/visual.ts

```diff
@@ -392,55 +392,89 @@ module powerbi.extensibility.visual {
             return result;
         }
 
-        private static getBrokenWords(
+        private static processText(
             words: WordCloudText[],
             stopWords: string[],
             settings: WordCloudSettings): WordCloudText[] {
-
-            const brokenStrings: WordCloudText[] = [],
+            let processedText: WordCloudText[] = [],
+                partOfProcessedText: WordCloudText[] = [],
                 whiteSpaceRegExp: RegExp = /\s/,
                 punctuationRegExp: RegExp = new RegExp(`[${WordCloud.Punctuation.join("\\")}]`, "gim");
 
-            if (!settings.general.isBrokenText) {
-                return words;
-            }
-
             words.forEach((item: WordCloudText) => {
                 if (typeof item.text === "string") {
                     let splittedWords: string[] = item.text
-                        .replace(punctuationRegExp, " ")
+                        .replace(punctuationRegExp, "")
                         .split(whiteSpaceRegExp);
+                    const splittedWordsOriginalLength: number = splittedWords.length;
 
-                    if (settings.stopWords.show) {
-                        splittedWords = splittedWords.filter((value: string) => {
-                            return value.length > 0 && !stopWords.some((removeWord: string) => {
-                                return value.toLocaleLowerCase() === removeWord.toLocaleLowerCase();
-                            });
-                        });
-                    }
-
-                    splittedWords.forEach((splittedWord: string) => {
-                        if (splittedWord.length === 0 || whiteSpaceRegExp.test(splittedWord)) {
-                            return;
-                        }
+                    splittedWords = WordCloud.getFilteredWords(splittedWords, stopWords, settings);
+                    partOfProcessedText = settings.general.isBrokenText
+                        ? WordCloud.getBrokenWords(splittedWords, item, whiteSpaceRegExp)
+                        : WordCloud.getFilteredSentences(splittedWords, item, splittedWordsOriginalLength);
 
-                        brokenStrings.push({
-                            text: splittedWord,
-                            textGroup: item.textGroup,
-                            count: item.count,
-                            index: item.index,
-                            selectionId: item.selectionId,
-                            color: item.color
-                        });
-                    });
+                    Array.prototype.push.apply(processedText, partOfProcessedText);
```

Comment:
I'm not sure that using ```apply``` is a good way. The ```processedText``` requires three arguments, but this call applies only the first argument. I'd prefer to use explicit call of the function/method instead.